edgedb-protocol/src/query_arg.rs

```diff
@@ -137,7 +135,7 @@ impl QueryArgs for () {
                 let root = enc.ctx.root_pos.and_then(|p| enc.ctx.get(p).ok());
                 match root {
                     Some(Descriptor::Tuple(t))
-                    if t.id == Uuid::from_u128(0xFF)
+                    if *t.id == Uuid::from_u128(0xFF)
```

Comment:
Can we also do `impl PartialEq<&Uuid> for DescriptorUuid` ?

Reply:
Okay, did impl PartialEq<Uuid> which gets rid of all the dereferencing there and in the check_descriptor function with all the assertions.